src/genpy/generate_fields.py

```diff
@@ -0,0 +1,26 @@
+from typing import List, Tuple
+import genmsg
+from genpy.base import PY_TYPE_STRINGS
+
+
+
+def generate_fields(spec_names: List[str], spec_types: List[str] ) -> List[Tuple[str, str, str]]:
+    field_definitions: List[str] = []
+    for spec_name, spec_type in zip(spec_names, spec_types):
+
+        base_type, is_array, array_length = genmsg.msgs.parse_type(spec_type)
+        if len(base_type.split('/')) > 1:
+            base_type = '.'.join(base_type.split('/')[:-1]) + '.msg.' +  base_type.split('/')[-1]
+        
+        if base_type in PY_TYPE_STRINGS:
+            base_type = PY_TYPE_STRINGS[base_type]
+    
+        if is_array and array_length:
+            format_spec_type_hint = f'Tuple[{", ".join([base_type]*array_length)}]'
```

Comment:
For large arrays with fixed length, I think this could get unwieldy - have you checked what it looks like for a large array?

Reply:
Theoretically you should never need to look at the generated method code so style is not super important

Reply:
Sorry this is confusing, which is a result of the the genmsg api not being written super nicely. `array_len` is not necessarily a number, if it can be a bool or None. So `is_array and array_length` basically saying this is a fixed length array so we want it to be typed as tuple. Dynamically lengthed arrays will be hinted like in the screenshot,.


REF: https://github.com/Pickle-Robot/genmsg/blob/da7534a7efd97b44437ac51cf09d5999d24a062c/src/genmsg/msgs.py#L88

Reply:
Wonderful - thanks for the explanation